File: training/src/training/yukicoder/no345/python/Main.py
def execute(arg):
    ans = -1
    cIndex = []
    wIndex = []

    for idx, s in enumerate(arg):
        if(s == "c"):
            cIndex.insert(0, idx)

        if(s == "w"):
            if(len(cIndex) > 0 and len(wIndex) > 0):
                wLast = wIndex[0]

                for cIdx in cIndex:
                    if(cIdx > wLast):
                        continue

                    length = idx - cIdx + 1
                    if(ans == -1 or ans > length):
                        ans = length
                        break
                
            wIndex.insert(0, idx)

    return ans


arg = "ilovechiwawa"
print(execute(arg))

File: training/src/training/yukicoder/no345/java/Main.java
package training.yukicoder.no345.java;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class Main {

	public static void main(String[] args) {
		String arg = "alkunytaicvumkyawegmrxukikxwytyercevcjunwaiearinininininc";

		int ret = new Main().execute(arg);
		System.out.println(ret);
	}

	private int execute(String arg) {
		List<Integer> cIndexs = findIndexs(arg, "c");
		List<Integer> wIndexs = findIndexs(arg, "w");

		List<Integer> chiwawaLengths = findChiwawaLengths(cIndexs, wIndexs);

		Optional<Integer> minLength = chiwawaLengths.stream().min((a, b) -> a.compareTo(b));
		return minLength.orElse(-1);
	}

	private List<Integer> findIndexs(String origin, String target) {
		List<Integer> ret = new ArrayList<>();
		int i = 0;
		while (origin.indexOf(target, i) > -1) {
			int index = origin.indexOf(target, i);
			ret.add(index);
			i = index + 1;
		}
		return ret;
	}

	private List<Integer> findChiwawaLengths(List<Integer> cIndexs, List<Integer> wIndexs) {
		List<Integer> ret = new ArrayList<>();
		for (Integer cIndex : cIndexs) {
			int count = 0;
			for (Integer wIndex : wIndexs) {
				if(cIndex.compareTo(wIndex) > 0) {
					continue;
				}

				if(++count == 2) {
					ret.add(wIndex - cIndex + 1);
					break;
				}
			}
		}
		return ret;
	}
	
}
